File: tests/fixtures/python/low_cohesion.py
"""Class with low cohesion — methods don't share state (future: LCOM4)."""


class KitchenSink:
    def __init__(self):
        self.users = []
        self.orders = []
        self.logs = []
        self.config = {}

    def add_user(self, user):
        self.users.append(user)

    def remove_user(self, user_id):
        self.users = [u for u in self.users if u.id != user_id]

    def get_user(self, user_id):
        return next((u for u in self.users if u.id == user_id), None)

    def create_order(self, order):
        self.orders.append(order)

    def cancel_order(self, order_id):
        self.orders = [o for o in self.orders if o.id != order_id]

    def get_order(self, order_id):
        return next((o for o in self.orders if o.id == order_id), None)

    def log_event(self, event):
        self.logs.append(event)

    def get_logs(self):
        return self.logs

    def clear_logs(self):
        self.logs = []

    def set_config(self, key, value):
        self.config[key] = value

    def get_config(self, key):
        return self.config.get(key)

File: tests/fixtures/typescript/low_cohesion.ts
class KitchenSink {
    private users: any[] = [];
    private orders: any[] = [];
    private logs: any[] = [];
    private config: any = {};

    addUser(user: any) { this.users.push(user); }
    removeUser(id: number) { this.users = this.users.filter(u => u.id !== id); }
    getUser(id: number) { return this.users.find(u => u.id === id); }

    createOrder(order: any) { this.orders.push(order); }
    cancelOrder(id: number) { this.orders = this.orders.filter(o => o.id !== id); }
    getOrder(id: number) { return this.orders.find(o => o.id === id); }

    logEvent(event: any) { this.logs.push(event); }
    getLogs() { return this.logs; }
    clearLogs() { this.logs = []; }

    setConfig(key: string, value: any) { this.config[key] = value; }
    getConfig(key: string) { return this.config[key]; }
}
